Firmware/ESP32/Hardware_bringup/lib/DEV/dev_battery.c

```diff
@@ -53,5 +65,85 @@ static inline void dev_battery_private_gpio_config(void)
 void dev_battery_init(void)
 {
     dev_battery_private_gpio_config();
+    adc2_config_channel_atten( ADC2_CHANNEL_2, ADC_ATTEN_DB_0 );
+}
+
+float dev_battery_convert_to_lipo_volts(int32_t raw_adc)
+{
```

Comment:
macro function instead:
```c
#define DEV_BATTERY_RAW_TO_VOLTAGE(raw_adc) (float)((raw_adc)/(4096) * ((DEV_BATTERY_PULLUP_KOHMS) + (DEV_BATTERY_PULLDOWN_KOHMS)) / (DEV_BATTERY_PULLDOWN_KOHMS))
```

Reply:
Try to have one division.
Cast raw_adc into int32_t